api/src/main/resources/liquibase-update-2.1.xml

```diff
@@ -0,0 +1,388 @@
+								 referencedColumnNames="user_id"/>
+		<addForeignKeyConstraint constraintName="concept_attribute_type_retired_by_fk"
+								 baseTableName="concept_attribute_type" baseColumnNames="retired_by"
+								 referencedTableName="users"
+								 referencedColumnNames="user_id"/>
+	</changeSet>
+
+	<changeSet id="20160427-0954-create-concept-attribute-table" author="bahmni">
+		<preConditions onFail="MARK_RAN">
+			<not>
+				<tableExists tableName="concept_attribute"/>
+			</not>
+		</preConditions>
+		<comment>Creating concept_attribute table</comment>
+		<createTable tableName="concept_attribute">
+			<column name="concept_attribute_id" type="int" autoIncrement="true">
+				<constraints primaryKey="true" nullable="false"/>
+			</column>
+			<column name="concept_id" type="int">
+				<constraints nullable="false"/>
+			</column>
+			<column name="attribute_type_id" type="int">
+				<constraints nullable="false"/>
+			</column>
+			<column name="value_reference" type="text">
+				<constraints nullable="false"/>
+			</column>
+			<column name="uuid" type="char(38)">
+				<constraints nullable="false" unique="true"/>
+			</column>
+			<column name="creator" type="int">
+				<constraints nullable="false"/>
+			</column>
+			<column name="date_created" type="datetime">
+				<constraints nullable="false"/>
+			</column>
+			<column name="changed_by" type="int"/>
+			<column name="date_changed" type="datetime"/>
+			<column name="voided" type="BOOLEAN" defaultValueBoolean="false">
+				<constraints nullable="false"/>
+			</column>
+			<column name="voided_by" type="int"/>
+			<column name="date_voided" type="datetime"/>
+			<column name="void_reason" type="varchar(255)" defaultValue="null"/>
+		</createTable>
+		<addForeignKeyConstraint constraintName="concept_attribute_concept_fk"
+								 baseTableName="concept_attribute" baseColumnNames="concept_id"
+								 referencedTableName="concept"
+								 referencedColumnNames="concept_id"/>
+		<addForeignKeyConstraint constraintName="concept_attribute_attribute_type_id_fk"
+								 baseTableName="concept_attribute" baseColumnNames="attribute_type_id"
+								 referencedTableName="concept_attribute_type"
+								 referencedColumnNames="concept_attribute_type_id"/>
+		<addForeignKeyConstraint constraintName="concept_attribute_creator_fk"
+								 baseTableName="concept_attribute" baseColumnNames="creator" referencedTableName="users"
+								 referencedColumnNames="user_id"/>
+		<addForeignKeyConstraint constraintName="concept_attribute_changed_by_fk"
+								 baseTableName="concept_attribute" baseColumnNames="changed_by"
+								 referencedTableName="users"
+								 referencedColumnNames="user_id"/>
+		<addForeignKeyConstraint constraintName="concept_attribute_voided_by_fk"
+								 baseTableName="concept_attribute" baseColumnNames="voided_by"
+								 referencedTableName="users"
+								 referencedColumnNames="user_id"/>
+	</changeSet>
+
+</databaseChangeLog>
```

Comment:
I had put changes prior to and including 2.1 into the file and changes after 2.1 into liquibase-update-to-latest.xml. Happy to reshuffle changes but I would need some help here what should go where. Can we have a quick chat some time so that I understand the constraints you mentioned better?